src/VecSim/spaces/L2_space.cpp

```diff
@@ -161,11 +162,18 @@ dist_func_t<float> L2_FP16_GetDistFunc(size_t dim, unsigned char *alignment, con
     auto features = (arch_opt == nullptr)
                         ? cpu_features::GetX86Info().features
                         : *static_cast<const cpu_features::X86Features *>(arch_opt);
+#ifdef OPT_AVX512_FP16
+    if (features.avx512_fp16 && dim >= 440) {
```

Comment:
Consider linking to the PR here (so we can see the graphs and have an explanation for the dimension condition